docs/SpellExample.java

```diff
@@ -0,0 +1,315 @@
+
+        //Encode the list of spells into an expression
+        Expr encodedSpellsList = Value(spellList);
+
+        //This query can be approximately read as for each spell in the list of spells evaluate the lambda function.
+        //That lambda function creates a temporary variable with each spell in the list and passes it to the create function.
+        //The create function then stores that spell in the database
+        Value spellsListSave = client.query(
+            Foreach(
+                encodedSpellsList,
+                Lambda(Value(NXT_SPELL),
+                    Create(
+                        Class(Value(SPELLS_CLASS)),
+                        Obj("data", Var(NXT_SPELL))
+                    )
+                )
+            )
+        ).get();
+
+        System.out.println("Created list of spells from java list: \n" + toPrettyJson(spellsListSave));
+        Collection<Spell> spellCollection = spellsListSave.asCollectionOf(Spell.class).get();
+        System.out.println("save " + spellCollection.size() + " spells:");
+        spellCollection.forEach(nextSpell -> System.out.println("   " + nextSpell));
+
+        System.out.println("\n");
+
+        /*
+         * Read all Spells for the Spells Index
+         */
+
+        //Lambda Variable for each spell ref
+        String REF_SPELL_ID = "NXT_SPELL";
+
+        //Map is equivalent to a functional map which maps over the set of all values returned by the paginate.
+        //Then for each value in the list it runs the lambda function which gets and returns the value.
+        //Map returns the data in a structure like this -> {"data": [ {"data": ...}, {"data": ...} ]} so the data field needs to be selected out.
+        //SelectAll does this by selecting the nested data with the Path("data", "data")
+        Value findAllSpells = client.query(
+            SelectAll(Path("data", "data"),
+                Map(
+                    Paginate(
+                        Match(Index(Value(INDEX_NAME)))
+                    ),
+                    Lambda(Value(REF_SPELL_ID), Get(Var(REF_SPELL_ID))))
+            )
+        ).get();
+
+        Collection<Spell> allSpellsCollection = findAllSpells.asCollectionOf(Spell.class).get();
+        System.out.println("read " + allSpellsCollection.size() + " spells:");
+        allSpellsCollection.forEach(nextSpell -> System.out.println("   " + nextSpell));
+
+        System.out.println("\n");
+
+        /*
+         * Just to keep things neat and tidy, delete the database and close the client connection
+         */
+        deleteDB(adminClient, DB_NAME);
+        System.out.println("Disconnected from FaunaDB as Admin!");
+
+        // add this at the end of execution to make things shut down nicely
+        System.exit(0);
+    }
+
+    private static void deleteDB(FaunaClient adminClient, String dbName) throws InterruptedException, java.util.concurrent.ExecutionException, JsonProcessingException {
```

Comment:
Maybe just throw `Exception`? The actual exception types don't seem relevant for the context.